### ISSUE TITLE
srun and sbatch/salloc do not operate as described in JADE documentation

### ISSUE THREAD

**Issue by MatthewWilletts** [jade-hpc-gpu/jade-hpc-gpu.github.io#90]
**TL:DR: The behaviour of the cluster does not match the documentation for JADE**

Previously this was marked as **_Critical_** on the Hartree Centre ServiceNow Portal for 11 days, but was subsequently closed with instructions to repost here.

**Issue:**

On http://docs.jade.ac.uk/en/latest/jade/scheduler/, in the table describing salloc and srun, we have the text: 

> salloc: allocate resources for a job in real time (typically used to allocate resources and spawn a shell, in which the srun command is used to launch parallel tasks)

So from this we expect to be able to have an salloc, or a sbatch script, that sets NTASKS to some value greater than one, and then have "srun ./some_exec.sh &" repeated, so as to have NTASKS running simultaneously. 

This is a stack overflow thread describing this: 
[Link](https://stackoverflow.com/questions/39186698/what-does-the-ntasks-or-n-tasks-does-in-slurm)

But running this on JADE, 'srun .... &' DOES NOT set off the srun sub-routines in parallel. 
Instead they are set off sequentially - only one srun task operates at any one time.

Running the jobs sruns as in the stackoverflow answer, two sleep commands of 10 and 12 seconds, takes 22 seconds +, doing them one after the other.

To see the output from doing this, run on JADE:
`'sacct -j 297795 --format=JobID,Start,End,Elapsed,NCPUS'`

**Comment by tomaslaz** [comment]
Please try the following:

Job submission file:

#!/bin/bash
#SBATCH --nodes=1
#SBATCH --gres=gpu:4
#SBATCH --partition=big
#SBATCH --ntasks=4
#SBATCH --nodes=1
# #SBATCH --cpus-per-task=1
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
srun -n1 --gres=gpu:1 execute.sh &
wait

execute.sh file:
#! /bin/bash
sleep 10 ; echo $(date +%x_%r)

Output:
25/01/19_12:25:21 pm GMT
25/01/19_12:25:21 pm GMT
25/01/19_12:25:21 pm GMT
25/01/19_12:25:21 pm GMT
srun: Job step creation temporarily disabled, retrying
srun: Job step created
25/01/19_12:25:31 pm GMT
25/01/19_12:25:31 pm GMT
25/01/19_12:25:31 pm GMT
25/01/19_12:25:31 pm GMT

As I understand the default behaviour is that if --gres is not set for each srun command it uses the one set for the job and every job (srun command) would reserve 4 gpus - in your case it leads to them being executed one by one.
